code/game/objects/items/RCD.dm

```diff
@@ -276,6 +402,123 @@ GLOBAL_VAR_INIT(icon_holographic_window, init_holographic_window())
 	. = ..()
 	AddElement(/datum/element/openspace_item_click_handler)
 
+	//Create the radial menu as a static list only at the beginning of the map for all RCD instances to share
+	if(!mapload)
+		return
+
+	var/list/icon_list = list()
+	var/list/essentials = list(
+		'icons/hud/radial.dmi' = list("wallfloor", "windowsize", "windowtype", "grillewindow", "cnorth", "csouth", "ceast", "cwest", "chair", "stool", "windoor", "secure_windoor", "airlock", "airlocktype", "access", "computer_dir"),
+		'icons/obj/structures.dmi' = list("window0", "rwindow0", "table", "glass_table"),
+		'icons/obj/stock_parts.dmi' = list("box_1"),
+		'icons/obj/mining.dmi' = list("silo"),
+		'icons/obj/doors/airlocks/station/public.dmi' = list("closed")
+	)
+	var/icon/icon
+	for(var/icon_file as anything in essentials)
+		for(var/icon_state as anything in essentials[icon_file])
+			icon = icon(icon = icon_file, icon_state = icon_state)
+			if(icon_state == "closed")
+				icon.Blend(icon(icon = icon_file, icon_state = "fill_closed", dir = SOUTH), ICON_OVERLAY)
+				icon_list["airlock-solid"] = icon
+
+				icon = icon(icon = icon_file, icon_state = icon_state)
+				icon_list["airlock-glass"] = icon
+			else
+				icon = icon(icon = icon_file, icon_state = icon_state)
+				if(icon_state == "window0" || icon_state == "rwindow0")
+					icon.Blend(icon(icon = 'icons/obj/structures.dmi', icon_state = "grille"), ICON_UNDERLAY)
+				icon_list[icon_state] = icon
+
+	//Initialize the radial menu
+	radial_menu = list(
+		"Main Menu" = list(
+			"Airlock" = list("icon" = icon_list["airlock"]),
+			"Grills & Windows" = list("icon" = icon_list["grillewindow"]),
+			"Floors & Walls" = list("icon" = icon_list["wallfloor"], "coordinates" = list("Root" = "Construction", "Category" = "Structures", "Index" = 1)),
+			"Machine Frame" = list("icon" = icon_list["box_1"], "requires" = RCD_UPGRADE_FRAMES, "coordinates" = list("Root" = "Construction", "Category" = "Machines", "Index" = 1)),
+			"Computer Frames" = list("icon" = icon_list["computer_dir"], "requires" = RCD_UPGRADE_FRAMES),
+			"Silo Link" = list("icon" = icon_list["silo"], "requires" = RCD_UPGRADE_SILO_LINK),
+			"Furnishing" = list("icon" = icon_list["chair"], "requires" = RCD_UPGRADE_FURNISHING),
+		),
+		"Grills & Windows" = list(
+			"Directional Window" = list("icon" = icon_list["windowsize"], "coordinates" = list("Root" = "Construction", "Category" = "Structures", "Index" = 2)),
```

Comment:
What is index and why are we typing them all manually? We will have to adjust them all if we want to change the order?

Reply:
These indices refer to the where the design is located in the static list `root_categories`. for example 
`"coordinates" = list("Root" = "Construction", "Category" = "Structures", "Index" = 2))`
actually refer to
`root_categories["Construction"]["Structures"][2] and that would be the design type which i call` ui_act()`. So yes we would have to type them manually but again this is the radial menu so we gonna scrap that?